### DISCUSSION TITLE
How to update?

### DISCUSSION THREAD

**Discussion by Bigfoot611** [LukeSkywalker92/teleframe#170] [Q&A] [answered]
Hi,

it's not really an issiue, but a question, so I've decided not to use issue tab:
Is there any kind of instruction or documentaton how to backup and restore data (picture database, allowed telegram sender, ...) when a major update is perfomed?
My frame is from Christmas 2019 and there are so many changes that I think it makes sense to reinstall, but I would very much like to keep the images and also the names and texts of the senders.

Thanks for any help or response...

**Comment by gegu** [comment, marked as answer]
If you use the [described update procedure](https://github.com/LukeSkywalker92/TeleFrame#-important-), everything should actually work.

To be safe, back up the `config` and `images` folders.

If you are reinstalling, follow these steps after installing and before starting TeleFrame for the first time - If TeleFrame  is already running, stop it first.:

- If the file `config/config.json` exists, delete it
- Copy the contents of the folders `config` and `images` from your backup into the newly created ones.

**Reply by Bigfoot611** [reply]
Thanks for your fast reply!
Update work like a charm... I really didn't expect this because I came from a really old version.

**Comment by gegu** [comment]
I forgot, should you update, the normal update procedure must still be performed afterwards 

```sh
git pull && npm install
```
